libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1807,26 +1802,15 @@ void TypeChecker::requireLValue(Expression const& _expression)
 
 void TypeChecker::typeError(SourceLocation const& _location, string const& _description)
```

Comment:
I could see replacing `warning( )`.  This refactor was a pretty iterative one, and based on the `typeError( )` and `fatalTypeError( )` calls, I saw some value in components having their own "choke points" to move all their error reporting through.  In the case of `warning( )` it's not necessarily all that useful, but it does help keep a symmetry in the code that can help with readability.